### ISSUE TITLE
False Positive: EmptyMethod

### ISSUE THREAD

**Issue by hzwaal** [scapegoat-scala/scapegoat#123]
Of course `scapegoat` is correct for the below example, but IMHO it should not report this if the method is implementing a `Unit` function.

``` Scala
  abstract class Thing {
    def method(): Unit
  }

  class DoNoThing extends Thing {
    // EmptyMethod: below code gives a warning that cannot be helped
    def method(): Unit = ()
  }
```

**Comment by mccartney** [comment]
Not thinking about it too much, but I guess it's more about the super class being abstract, not that the return type is `Unit`.

**Comment by Johnnei** [comment]
This case was fixed as a side effect of ffcc2ccb9f7cb692c95451126890733ab0a2d290. That commit made it only warn on private methods which abstract methods can't be.
